proof/crefine/AARCH64/ADT_C.thy

```diff
@@ -820,17 +844,21 @@ lemma cthread_state_rel_imp_eq:
   apply (cases y, simp_all add: ThreadState_defs)+
   done
 
-lemma ksPSpace_valid_objs_tcbBoundNotification_nonzero:
-  "\<exists>s. ksPSpace s = ah \<and> no_0_obj' s \<and> valid_objs' s
-     \<Longrightarrow> map_to_tcbs ah p = Some tcb \<Longrightarrow> tcbBoundNotification tcb \<noteq> Some 0"
+(* FIXME: choose better name for this lemma *)
```

Comment:
The assumption order is unusual, it seems like the characteristic assumption is the `map_to_tcbs` one, i.e. if you found some tcb at `p`, then its bound pointers can't be NULL pointers, and I'd put that first. Then the name would be something like `map_to_tcbs_Some_bound_objs_nonzero`. I don't think there will be another `bound_objs_nonzero` that is stronger than using `valid_objs'`

Reply:
Can do